carbon-design-system/carbon
# 10957 fix datepicker unknown prop warning

Closes #10957 

Adds `onOpen` to destructuring clearing the 'unknown prop' error in React

#### Testing / Reviewing

1. pull down the PR locally
2. in `packages/react/src/DatePicker/DatePicker-story.js` 
3. in the Single With Calendar story add `onOpen={() => console.log('open')}`
4. see the console log and you should NOT see a Uknown prop error from React
